actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -165,15 +167,29 @@ module CacheHelper
       # expire the cache.
       def cache(name = {}, options = {}, &block)
         if controller.respond_to?(:perform_caching) && controller.perform_caching
-          name_options = options.slice(:skip_digest)
-          safe_concat(fragment_for(cache_fragment_name(name, **name_options), options, &block))
+          begin
+            caching_was = ActionView::CachingRegistry.is_caching
+            ActionView::CachingRegistry.is_caching = true
+
+            name_options = options.slice(:skip_digest)
+            safe_concat(fragment_for(cache_fragment_name(name, **name_options), options, &block))
+          ensure
+            ActionView::CachingRegistry.is_caching = caching_was
+          end
```

Comment:
The `CacheHelper` shouldn't track the registry state manually, let's extract a helper:

```suggestion
          ActionView::CachingRegistry.track_caching do
            name_options = options.slice(:skip_digest)
            safe_concat(fragment_for(cache_fragment_name(name, **name_options), options, &block))
          end
```